openant/devices/shift.py

```diff
@@ -99,10 +140,60 @@ def on_data(self, data):
             self.data["shift"].shift_failure_front = (data[7] & 0xF0) >> 4
 
             delta_update_count = (
-                self._event_count[1] + 256 - self._event_count[0]
+                self._event_count[0][1] + 256 - self._event_count[0][0]
             ) % 256
 
             # if it's a new event (count change)
             if delta_update_count:
                 _logger.info(f"Shifting status update {self}: {self.data['shift']}")
                 self.on_device_data(page, "shift_system_status", self.data["shift"])
+        elif page == 0x02:
+            self._event_count[1][0] = self._event_count[1][1]
+            self._event_count[1][1] = data[1]
+
+            self.data["shift"].event_count = data[1]
+
+            self.data["shift"].event_1 = FunctionSetEvent(data[2] & 0x0F, (data[2] & 0x03) >> 4)
```

Comment:
The second parameter needs to be wrapped with `FunctionSetEventType` to cast as the Enum. Even with the typing Python doesn't auto cast so this will be a int in the dataclass currently.

Reply:
Now correctly cast